src/main/java/uk/org/tombolo/importer/phe/ChildhoodObesityImporter.java

```diff
@@ -98,113 +72,181 @@ public DatasourceSpec getDatasourceSpec(String datasourceId) throws Exception {
 
     @Override
     protected void importDatasource(Datasource datasource, List<String> geographyScope, List<String> temporalScope,  List<String> datasourceLocation) throws Exception {
-        // Choose the apppropriate workbook sheet
-        Workbook workbook = excelUtils.getWorkbook(
-                downloadUtils.fetchInputStream(new URL(DATASOURCE), getProvider().getLabel(), DATASOURCE_SUFFIX)
-        );
-        Sheet sheet = null;
-        SubjectType subjectType = null;
-        String year = "2014";
-        for (String geographyScopeString : geographyScope) {
-            GeographyLabel geographyLabel = GeographyLabel.valueOf(geographyScopeString);
-            switch (geographyLabel) {
-                case la:
-                    sheet = workbook.getSheet("LAData_2011-12_2013-14");
-                    subjectType = SubjectTypeUtils.getOrCreate(AbstractONSImporter.PROVIDER,
-                            OaImporter.OaType.localAuthority.name(), OaImporter.OaType.localAuthority.datasourceSpec.getDescription());
-                    break;
-                case msoa:
-                    sheet = workbook.getSheet("MSOAData_2011-12_2013-14");
-                    subjectType = SubjectTypeUtils.getOrCreate(AbstractONSImporter.PROVIDER,
-                            OaImporter.OaType.msoa.name(), OaImporter.OaType.msoa.datasourceSpec.getDescription());
-                    break;
-                case ward:
-                    throw new Error("Wards are not yet supported");
-                    // FIXME: In case we want to suport wards at some point, here is the sheet to use
-                    //sheet = workbook.getSheet("WardData_2011-12_2013-14");
-                    //break;
+
+        // Check the datasourceid to reference the corresponding url
+        String which_datasource = datasource.getDatasourceSpec().getId();
+        String fileLocation = getDatasourceSpec(which_datasource).getUrl();
+
+        List<TimedValue> timedValues = new ArrayList<TimedValue>();
+
+        // The code below fetches the .xls file from the URL we specified in our DatasourceSpec object
+        URL url;
+        try {
+            url = new URL(fileLocation);
+        } catch (MalformedURLException e) {
+            File file;
+            if (!(file = new File(fileLocation)).exists()) {
+                log.error("File does not exist: " + fileLocation);
             }
-            if (sheet == null)
-                throw new Error("Sheet not found for datasource: " + datasource.getDatasourceSpec().getId());
+            url = file.toURI().toURL();
+        }
+
+        // Fetching and reading the file using fetchInputStream
+        InputStream isr = downloadUtils.fetchInputStream(url, getProvider().getLabel(), ".xlsx");
+        XSSFWorkbook workbook = new XSSFWorkbook(isr);
+        DataFormatter dataFormatter = new DataFormatter();
+
```

Comment:
i didn't see this object being used anywhere, if it is not used please delete it

Reply:
Done